kong/api/routes/upstreams.lua

```diff
@@ -160,4 +103,25 @@ return {
     end
   },
 
+  ["/upstreams/:name_or_id/targets/:target"] = {
+    before = function(self, dao_factory, helpers)
+      crud.find_upstream_by_name_or_id(self, dao_factory, helpers)
+      self.params.upstream_id = self.upstream.id
+    end,
+
+    DELETE = function(self, dao_factory)
+      balancer.clean_history(self.params.upstream_id, dao_factory)
+
+      -- this is just a wrapper around POSTing a new target with weight=0
+      local target  = self.params
```

Comment:
Hmm? `self.params` represents the target (a table with a  `target` key), not an upstream.